File: db4on/core/src/core/com/db4o/internal/query/processor/QEGreater.cs
namespace com.db4o.@internal.query.processor
{
	/// <exclude></exclude>
	public class QEGreater : com.db4o.@internal.query.processor.QEAbstract
	{
		internal override bool Evaluate(com.db4o.@internal.query.processor.QConObject a_constraint
			, com.db4o.@internal.query.processor.QCandidate a_candidate, object a_value)
		{
			if (a_value == null)
			{
				return false;
			}
			return a_constraint.GetComparator(a_candidate).IsGreater(a_value);
		}

		public override void IndexBitMap(bool[] bits)
		{
			bits[com.db4o.@internal.query.processor.QE.GREATER] = true;
		}
	}
}

File: db4oj/core/src/com/db4o/QEGreater.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

public class QEGreater extends QEAbstract
{
	boolean evaluate(QConObject a_constraint, QCandidate a_candidate, Object a_value){
		if(a_value == null){
			return false;
		}
		return a_constraint.getComparator(a_candidate).isGreater(a_value);
	}
	
	void indexBitMap(boolean[] bits){
	    bits[2] = true;
	}
}
